Lib/unittest/mock.py

```diff
@@ -3004,13 +3004,32 @@ def __get__(self, obj, obj_type=None):
     def __set__(self, obj, val):
         self(val)
 
+
+_timeout_unset = sentinel.TIMEOUT_UNSET
+
 class ThreadingMixin(Base):
 
-    def __init__(self, *args, **kwargs):
+    DEFAULT_TIMEOUT = _timeout_unset
+
+    def _get_child_mock(self, /, **kw):
+        if "timeout" in kw:
+            kw["timeout"] = kw.pop("timeout")
+        elif isinstance(kw.get("parent"), ThreadingMixin):
+            kw["timeout"] = kw["parent"]._mock_wait_timeout
```

Comment:
Do we have a test that verifies this? I think it's important functionality :-)

Reply:
Yup, note all tests that invoke `m.method1`. Calling that generates a child mock.
Happy to add more tests if there is anything else you want covered.